File: Go/add_two_numbers/2_test.go
package addtwonumbers

import (
	"fmt"
	"testing"

	"github.com/stretchr/testify/assert"
)

func Test_addTwoNumbers(t *testing.T) {
	tests := []struct {
		l1   *ListNode
		l2   *ListNode
		want *ListNode
	}{
		{
			l1:   &ListNode{2, &ListNode{4, &ListNode{3, nil}}},
			l2:   &ListNode{5, &ListNode{6, &ListNode{4, nil}}},
			want: &ListNode{7, &ListNode{0, &ListNode{8, nil}}},
		},
		{
			l1:   &ListNode{0, nil},
			l2:   &ListNode{0, nil},
			want: &ListNode{0, nil},
		},
		{
			l1:   &ListNode{9, &ListNode{9, &ListNode{9, &ListNode{9, &ListNode{9, &ListNode{9, &ListNode{9, nil}}}}}}},
			l2:   &ListNode{9, &ListNode{9, &ListNode{9, &ListNode{9, nil}}}},
			want: &ListNode{8, &ListNode{9, &ListNode{9, &ListNode{9, &ListNode{0, &ListNode{0, &ListNode{0, &ListNode{1, nil}}}}}}}},
		},
	}

	for i, tt := range tests {
		t.Run(fmt.Sprintf("case %d", i+1), func(t *testing.T) {
			got := addTwoNumbers(tt.l1, tt.l2)
			assert.Equal(t, tt.want, got)
		})
	}
}

File: TypeScript/add_two_numbers/2.test.ts
import { assertEquals } from "@std/assert";
import { addTwoNumbers, ListNode } from "./2.ts";

interface tt {
  l1: ListNode;
  l2: ListNode;
  want: ListNode;
}

const cases: tt[] = [
  {
    l1: new ListNode(2, new ListNode(4, new ListNode(3))),
    l2: new ListNode(5, new ListNode(6, new ListNode(4))),
    want: new ListNode(7, new ListNode(0, new ListNode(8))),
  },
  {
    l1: new ListNode(
      9,
      new ListNode(
        9,
        new ListNode(
          9,
          new ListNode(9, new ListNode(9, new ListNode(9, new ListNode(9)))),
        ),
      ),
    ),
    l2: new ListNode(9, new ListNode(9, new ListNode(9, new ListNode(9)))),
    want: new ListNode(
      8,
      new ListNode(
        9,
        new ListNode(
          9,
          new ListNode(
            9,
            new ListNode(0, new ListNode(0, new ListNode(0, new ListNode(1)))),
          ),
        ),
      ),
    ),
  },
];

for (const t of cases) {
  Deno.test(JSON.stringify(t), () => {
    const result = addTwoNumbers(t.l1, t.l2);
    assertEquals(result, t.want);
  });
}
